core/src/syscall/write.rs

```diff
@@ -72,3 +78,17 @@ impl Syscall for SyscallWrite {
         0
     }
 }
+
+pub fn update_io_buf(ctx: &mut SyscallContext, fd: u32, s: &str) -> Option<String> {
+    let rt = &mut ctx.rt;
+    if s.ends_with('\n') {
```

Comment:
I think this won't print immediately in some cases like if you do print!("hi\nhi")

I think the logic should be: add s to the buf, then split buf by \n, print each line except last, and leave last line in the buf

Also, the runtime needs to flush anything remaining in the buf at the end of runtime